examples/contracts/examples/generate.rs

```diff
@@ -0,0 +1,39 @@
+//! Example of generating code from ABI file to interact with the contract.
+
+use alloy_network::Ethereum;
+use alloy_node_bindings::Anvil;
+use alloy_provider::{ProviderBuilder, RootProvider};
+use alloy_rpc_client::RpcClient;
+use alloy_sol_types::sol;
+use alloy_transport_http::Http;
+use eyre::Result;
+use reqwest::Client;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    // Codegen from ABI file to interact with the contract.
+    sol!(
+        #[sol(rpc)]
+        IERC20,
+        "examples/abi/IERC20.json"
+    );
+
+    // Spin up a forked Anvil node.
+    // Ensure `anvil` is available in $PATH
+    let anvil = Anvil::new().fork("https://eth.llamarpc.com").spawn();
+
+    // Create a provider.
+    let http = Http::<Client>::new(anvil.endpoint().parse()?);
+    let provider = ProviderBuilder::<_, Ethereum>::new()
+        .provider(RootProvider::new(RpcClient::new(http, true)));
+
+    // Create a contract instance.
+    let contract = IERC20::new("0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2".parse()?, provider);
+
+    // Call the contract, retrieve the total supply.
+    let IERC20::totalSupplyReturn { _0 } = contract.totalSupply().call().await?;
```

Comment:
still think we should do snake_case and add serde-renames style attributes to the sol macro  @DaniPopes @prestwich

Reply:
before we add any more features to `sol!` we need to refactor to break out the input and expanders to make it more maintainable

Reply:
Created a ticket here: https://github.com/alloy-rs/alloy/issues/357 to continue the conversation

Going to merge this ticket as is - will reflect any changes in Alloy here